src/SourceBuild/content/build.proj

```diff
@@ -43,6 +53,132 @@
       VmrBuildNumber="$(BUILD_BUILDNUMBER)" />
   </Target>
 
+  <PropertyGroup>
+    <ExtractDotNetSdkArchiveDependsOn>SetupScenarioTestsArtifacts;GetDotNetSdkArtifactPath</ExtractDotNetSdkArchiveDependsOn>
+    <ExtractDotNetSdkArchiveDependsOn Condition="'$(ArchiveExtension)' == '.zip'">$(ExtractDotNetSdkArchiveDependsOn);ExtractDotNetSdkZip</ExtractDotNetSdkArchiveDependsOn>
+    <ExtractDotNetSdkArchiveDependsOn Condition="'$(ArchiveExtension)' == '.tar.gz'">$(ExtractDotNetSdkArchiveDependsOn);ExtractDotNetSdkTarball</ExtractDotNetSdkArchiveDependsOn>
+  </PropertyGroup>
+
+  <Target Name="ExtractDotNetSdkArchive"
```

Comment:
The dotnet sdk layout is already available in the installer repo. Can we just copy that into a well-known location before the repo gets cleaned in installer.proj?

Reply:
I think it's better to consume the actual output rather than bypassing that. It gives better test coverage in case we didn't produce the archive file correctly.

Reply:
I'm fine to keep using the archive but be warry of executable permissions. Anyways, this logic should be moved into installer.proj. I want to keep the build.proj entry point file as minimal as possible. We will also eventually move the Publish target out of build.proj as it doesn't belong there.

Reply:
I don't think this should be in installer.proj either. Perhaps another project altogether?

Reply:
> The dotnet sdk layout is already available in the installer repo. Can we just copy that into a well-known location before the repo gets cleaned in installer.proj?

We should base our testing on the actual produced output, rather than taking a shortcut.

Reply:
> We should base our testing on the actual produced output, rather than taking a shortcut.

Good point, yes that makes sense.

> I don't think this should be in installer.proj either. Perhaps another project altogether?

There are multiple projects that needs the produced and extracted sdk archive: smoke tests, unified build tests, scenario tests and some source-build specific symbol archiving logic. On the other hand, short stack builds don't need it. I think installer.proj is suitable here as it produces that archive and should be responsible for then extracting it and that doesn't run for short stack builds.

Reply:
> source-build specific symbol archiving logic

Can you point to where this uses the extracted SDK?

> I think installer.proj is suitable here as it produces that archive and should be responsible for then extracting it and that doesn't run for short stack builds.

In terms of separation of concerns, it seems to me that extracting the SDK is not part of the build. It's only a necessary step because of testing. So doing it by default in a build is unnecessary and should be left to the test logic.

Reply:
If extracting the SDK would only be necessary for testing, then it shouldn't be done as part of the build, correct. But that's not the case as creating the sdk symbol archiving already needs the extracted layout: https://github.com/dotnet/installer/blob/144a405fd5fe407d34cf86ed4e68ec34268eca5b/src/SourceBuild/content/eng/build.sourcebuild.targets#L46